mantle/platform/qemu.go

```diff
@@ -306,6 +306,41 @@ func (inst *QemuInstance) SwitchBootOrder() (err error) {
 	return nil
 }
 
+// RemoveBlockDevice deletes the primary device from a qemu instance
+// and sets the seconday device as primary.
+func (inst *QemuInstance) RemoveBlockDevice() error {
+	var primaryDevice string
+	var secondaryDevicePath []string
+	monitor, err := newQMPMonitor(inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+	monitor.Connect()
+	defer monitor.Disconnect()
```

Comment:
Should probably mimic the `defer` @ https://github.com/coreos/coreos-assembler/blob/master/mantle/platform/qemu.go#L257-L264